tools/src/main/java/org/apache/kafka/tools/ProducerPerformance.java

```diff
@@ -114,6 +133,10 @@ record = new ProducerRecord<>(config.topicName, payload);
             if (!config.shouldPrintMetrics) {
                 producer.close();
 
+                /* print warmup stats if relevant */
+                if (warmupStats != null) {
+                    new Stats(warmupStats, stats).printTotal();
```

Comment:
Maybe we should have a state for 'all' and another state for 'steady-only'. With this approach, we can avoid this kind of magic merging.

Reply:
Thanks for the review @chia7712!
Just to be clear, are you suggesting that, instead of using a warmup stats followed by a steady state stats object, we should use a whole-test stats object in addition to a steady-state stats object?
I don't see any serious issues in the implementation, but I the producer callback will need refactoring so it writes into two stats objects once steady state has triggered. In the initial design, I was trying to avoid this concurrent need for both stats objects and do the array-merging at the end, but I will give it a try to see how it looks.

Reply:
@matt-welch did you tried the suggested approach?

Reply:
Sorry for the delay.  I've refactored the stats objects as suggested in my latest commit.